src/core/thread.d

```diff
@@ -2090,17 +2090,28 @@ extern (C) void thread_init()
         status = sem_init( &suspendCount, 0, 0 );
         assert( status == 0 );
     }
-    Thread.sm_main = thread_attachThis();
+    if (typeid(Thread).initializer.ptr)
+        _mainThreadStore[] = typeid(Thread).initializer[];
+    Thread.sm_main = attachThread((cast(Thread)_mainThreadStore.ptr).__ctor());
 }
 
+private __gshared align(Thread.alignof) void[__traits(classInstanceSize, Thread)] _mainThreadStore;
+
+extern (C) void _d_monitordelete_nogc(Object h) @nogc;
 
 /**
  * Terminates the thread module. No other thread routine may be called
  * afterwards.
  */
-extern (C) void thread_term()
+extern (C) void thread_term() @nogc
 {
-    destroy(Thread.sm_main);
+    assert(_mainThreadStore.ptr is cast(void*) Thread.sm_main);
+
+    // destruct manually as object.destroy is not @nogc
+    Thread.sm_main.__dtor();
+    _d_monitordelete_nogc(Thread.sm_main);
+    if (typeid(Thread).initializer.ptr)
+        _mainThreadStore[] = typeid(Thread).initializer[];
```

Comment:
A couple comments here:

My coverage analysis is saying this code is never run.

The normal destroy will nullify the vtable, and also if the `initializer.ptr` is null, it means you should zero the data. But I don't actually think you need to check if it's null, as it should always have a vtable in the initializer.

Reply:
Coverage messurement fails here because this is run outside of the main loop, i.e. after coverage is collected and reported.

Reply:
BTW, you can easily fixup or edit PRs. Much less hassle than asking for single-line changes.